src/iosrv/vos.c

```diff
@@ -646,10 +646,11 @@ close_iod(struct dss_enum_unpack_io *io)
 	D_ASSERTF(io->ui_iods_size <= io->ui_iods_cap, "%d < %d\n",
 		  io->ui_iods_size, io->ui_iods_cap);
 
-	if (io->ui_iods_size == 0)
+	if (io->ui_iods_size == 0 &&
+	    (io->ui_iods == NULL || io->ui_iods[0].iod_nr == 0))
```

Comment:
Is it possible that the @ui_iods_size is zero, but iod_nr is not zero?

Reply:
well, we are using ui_iods_size as the iod index as well. so ui_iods_size = 0 might not mean the empty iod. It is indeed a bit confusing, and  I will fix it in the following patch.